src/detect-engine-alert.c

```diff
@@ -280,8 +279,10 @@ int PacketAlertAppend(DetectEngineThreadCtx *det_ctx, const Signature *s,
                        ") in packet %" PRIu64 "",
                     p->alerts.alerts[i].s->id, p->alerts.alerts[i].num, s->id, s->num, p->pcap_cnt);
             PacketAlertInsertPos(det_ctx, s, p, tx_id, flags, i);
-            // TODO log to stats signature id that was discarded/replaced
         }
+        /* Update so we can log to stats later. If we got here, some alert was
+        discarded */
+        p->alerts.discarded++;
```

Comment:
It seems to me that we have at least two situations:
- a signature in the alert queue is replaced;
- the new match happens for a signature with a lower priority id, so it will be discarded and nothing changes in the alert queue.
Do we want to log these two as the same thing in the stats log?
If yes, then I think this patch is enough. If not, then more work has to be done.